src/app/view/components/_common/dot-page-selector/dot-page-selector.component.spec.ts

```diff
@@ -0,0 +1,175 @@
+        de = hostDe.query(By.css('dot-page-selector'));
+        component = de.componentInstance;
+        dotPageSelectorService = de.injector.get(DotPageSelectorService);
+
+        spyOn(component.selected, 'emit');
+        spyOn(dotPageSelectorService, 'getPagesInFolder').and.callThrough();
+        spyOn(component, 'writeValue').and.callThrough();
+
+        hostFixture.detectChanges();
+        autocomplete = de.query(By.css('p-autoComplete'));
+        autocompleteComp = autocomplete.componentInstance;
+
+    });
+
+    it('should have autocomplete', () => {
+        expect(autocomplete).toBeTruthy();
+    });
+
+    it('shold not set floating label directive', () => {
+        expect(de.query(By.css('[dotMdInputtext]')) === null).toBe(true);
+    });
+
+    it('should search for pages', () => {
+        autocomplete.triggerEventHandler('completeMethod', {
+            query: 'hello'
+        });
+
+        expect(dotPageSelectorService.getPagesInFolder).toHaveBeenCalledWith('hello');
+    });
+
+    it('should pass attritubes', () => {
+        expect(autocompleteComp.style).toEqual({'width': '100%'});
+        expect(autocompleteComp.placeholder).toEqual('Hello World');
+    });
+
+    describe('ControlValueAccessor', () => {
+        beforeEach(() => {
+            spyOn(component, 'propagateChange').and.callThrough();
+        });
+
+        it('should emit selected item and propagate changes', () => {
+
+            autocomplete.triggerEventHandler('onSelect', {identifier: '123'});
+            expect(component.selected.emit).toHaveBeenCalledWith({identifier: '123'});
+            expect(component.propagateChange).toHaveBeenCalledWith('123');
+        });
+
+        it('should write value', () => {
+            expect(component.writeValue).toHaveBeenCalledWith('f9fc55e7-557a-4047-a8be-15e5ca69fa62');
+            expect(component.val.identifier).toEqual('f9fc55e7-557a-4047-a8be-15e5ca69fa62');
+        });
+
+        it('should clear model and suggections', () => {
+            autocomplete.triggerEventHandler('onClear', {});
+            expect(component.propagateChange).toHaveBeenCalledWith(null);
+            expect(component.results).toEqual([]);
+        });
+    });
+});
+
+@Component({
+    selector: 'dot-fake-form',
+    template: `
+        <form [formGroup]="form">
+            <dot-page-selector
+                formControlName="page"
+                [style]="{'width': '100%'}"
+                label="Hello World"
+                [floatingLabel]="true">
+            </dot-page-selector>
+        </form>
+    `
+})
+class FakeForm2Component {
+    form: FormGroup;
+
+    constructor(private fb: FormBuilder) {
+        /*
+            This should go in the ngOnInit but I don't want to detectChanges everytime for
+            this fake test component
+        */
+        this.form = this.fb.group({
+            page: [{ value: 'f9fc55e7-557a-4047-a8be-15e5ca69fa62', disabled: false }]
+        });
+    }
+}
```

Comment:
I think is better add a attribute 'floatingLabel' for this componente and just set it to false for the rest of the test